### PR TITLE
box/lua: return exact nil from calls instead of empty return

### LINKED ISSUE #3716: Return null in case space:get doesn't find anything
`space:get()` returns tuple if it is found and nothing otherwise.
This is inconsistent and affects functions which pass the result of `get` function directly to a
c written code.

E.g. console prints
```
---
```
instead of 
```
---
- null
```

### PR DESCRIPTION
**Pull request by Mons**
Fix `luaT_pushtupleornil` and particular _ffi implementations
to return one nil argument instead of empty to change behaviour
of non-returning calls, like `:get`, `:update` or `:delete`
on non-existing keys.

Closes #3716

This patch changes behaviour of builtin functions.

The change is almost not seen from application point of view:
for almost any code there is no difference between
`return` and `return nil`, except for `select('#',...)`

But this change is visible in console and during yaml serialization:
Instead of empty yaml response
```
---
...
```
the response will be
```
---
- null
...
```

So, this affects many of test_run result files in such a way

### CODE REVIEW

File: src/box/lua/tuple.h [hunk 1]
```diff
@@ -33,6 +33,12 @@
 #include <stddef.h>
 #include "trivia/util.h"
 
+#include "lua/utils.h"
+#include "lua/trigger.h"
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
```
Comment (Totktonada) [inline src/box/lua/tuple.h:40]: Nits:

1. I guess `lua.h` is enough.
2. System headers should go first.

**Comment by Totktonada** [comment]
I should ask, you know: what motivates the change?

When we introduce a subtle behaviour difference (which may obligate a user to update some its code), we should give something for him / her: a bugfix or some new functionality.

Please, take me right: I don't fight against the change and, to be honest, I don't know whether we should proceed with it. I need to think around and the first thing I need is a clear reason, why you think that it worth doing.

File: src/box/lua/tuple.h (hunk 1 above)
Comment (Mons) [reply src/box/lua/tuple.h:40]: I can't get it to work if I remove some of the includes or change their order. Linking (!) fails with a lot of errors like
```lua
undefined reference to `lua_settop(lua_State*, int)'
```

**Comment by kyukhin** [comment]
No answer for 3+ months.

**Pull request closed without merging**